### PR TITLE
GG-19271

### PR DESCRIPTION
**Pull request by antonovsergey93**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 1]
```diff
@@ -121,21 +119,29 @@
...
-                    }
-                    else {
-                        if (new File(nextArg).isDirectory())
-                            filePath = nextArg;
+                            allNodes = true;
+                            break;
+                        case "--nodes":
+                            argIter.nextArg("");
+
+                            nodeIds.addAll(argIter.nextStringSet(nextArg));
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: Empty line missed

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 2]
```diff
@@ -121,21 +119,29 @@
...
+                        case "--all":
+                            argIter.nextArg("");
 
-                            nodeIds.add(nextArg);
-                        }
-                    }
-                    else {
-                        if (new File(nextArg).isDirectory())
-                            filePath = nextArg;
+                            allNodes = true;
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: Empty line missed

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 3]
```diff
@@ -2599,29 +2602,85 @@ public void testDiagnosticPageLocksTracker() throws Exception {
...
         ignite.cluster().active(true);
 
-        String dir = U.defaultWorkDirectory() + "/diagnostic/";
+        File default0 = new File(U.defaultWorkDirectory() + "/" + DEFAULT_TARGET_FOLDER + "/");
+        File testDir = new File(U.defaultWorkDirectory() + "/diagnostic_test_dir/");
 
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "help"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "help"));
+        if (default0.exists())
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: You should move cleaning files/directories from test to `befoteTest()`/`afterTest()` methods

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 4]
```diff
@@ -2599,29 +2602,85 @@ public void testDiagnosticPageLocksTracker() throws Exception {
...
+                U.delete(testDir);
+            }
+        }
+    }
+
+    /**
+     * @param dir Directory.
+     * @param numberFiles Number of files.
+     */
+    private void checkNumberFiles(File dir, int numberFiles) {
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: Replace to lambda please
```
File[] files = dir.listFiles((d, name) -> name.startsWith(ToFileDumpProcessor.PREFIX_NAME));
```

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 5]
```diff
@@ -2599,29 +2602,85 @@ public void testDiagnosticPageLocksTracker() throws Exception {
...
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+                "--path", testDir.getAbsolutePath(),
+                "--nodes", node1.consistentId().toString() + "," + node3.consistentId().toString()));
+
+            checkNumberFiles(testDir, 7);
+
+            success = true;
+        }
+        finally {
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: You should move cleaning files/directories from test to `befoteTest()`/`afterTest()` methods

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 6]
```diff
@@ -121,21 +119,29 @@
...
+                            break;
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: Empty line missed

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 7]
```diff
@@ -121,21 +119,29 @@
...
+                        case "--nodes":
+                            argIter.nextArg("");
+
+                            nodeIds.addAll(argIter.nextStringSet(nextArg));
+                            break;
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: Empty line missed

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 8]
```diff
@@ -121,21 +119,29 @@
...
+                        case "--path":
+                            argIter.nextArg("");
+
+                            filePath = argIter.nextArg("").toLowerCase();
+
+                            break;
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: Empty line missed

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 9]
```diff
@@ -129,6 +132,30 @@
...
+    private File defaultDiagnosticDir;
+    /** */
+    private File customDiagnosticDir;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        defaultDiagnosticDir = new File(U.defaultWorkDirectory() + "/" + DEFAULT_TARGET_FOLDER + "/");
+        customDiagnosticDir = new File(U.defaultWorkDirectory() + "/diagnostic_test_dir/");
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: I think we could merge field`s definition and field`s initialization to single command.

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 10]
```diff
@@ -2599,29 +2626,58 @@ public void testDiagnosticPageLocksTracker() throws Exception {
...
+
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log", "--all"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", dir, "--all"));
+        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+            "--path", customDiagnosticDir.getAbsolutePath(), "--all"));
+
+        checkNumberFiles(customDiagnosticDir, 5);
+
+        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+            "--nodes", node0.id().toString() + "," + node2.id().toString()));
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: Your could use for CommandLogger#join() method for joining objects with specified delimeter

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 11]
```diff
@@ -2599,29 +2626,58 @@ public void testDiagnosticPageLocksTracker() throws Exception {
...
+        checkNumberFiles(customDiagnosticDir, 1);
 
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--all"));
+
+        checkNumberFiles(defaultDiagnosticDir, 5);
+
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log", "--all"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", dir, "--all"));
+        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+            "--path", customDiagnosticDir.getAbsolutePath(), "--all"));
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: Hereinafter code style violation.
Must be
```
            assertEquals(
                EXIT_CODE_OK, 
                execute("--diagnostic", "pageLocks", "dump", "--path", testDir.getAbsolutePath(), "--all")
            );
```

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 12]
```diff
@@ -2599,29 +2626,58 @@ public void testDiagnosticPageLocksTracker() throws Exception {
...
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump"));
+
+        checkNumberFiles(defaultDiagnosticDir, 1);
+
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", dir));
+        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--path",
+            customDiagnosticDir.getAbsolutePath()));
+
+        checkNumberFiles(customDiagnosticDir, 1);
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: Why is there only one diagnostic file? It's not clear. Please add comments for all method invocation.

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/DiagnosticCommand.java [hunk 13]
```diff
@@ -101,20 +103,28 @@
...
     @Override public void printUsage() {
-        logger.info("View diagnostic information in a cluster. For more details type:");
-        logger.info(join(" ", UTILITY_NAME, DIAGNOSTIC, HELP));
-        logger.info("\n");
+        Command.usage("View diagnostic information in a cluster:", DIAGNOSTIC);
     }
 
     /**
      *
      */
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/DiagnosticCommand.java]: Single line comment must be here

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/DiagnosticCommand.java [hunk 14]
```diff
@@ -101,20 +103,28 @@
...
     private void printDiagnosticHelp() {
-        logger.info(join(" ", UTILITY_NAME, DIAGNOSTIC, PAGE_LOCKS + " - dump page locks info."));
+        CommandLogger.logWithIndent(join(" ", UTILITY_NAME, DIAGNOSTIC, PAGE_LOCKS + " - dump page locks info."));
+
+        CommandLogger.nl();
+
+        CommandLogger.logWithIndent("Subcommands:");
+
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/DiagnosticCommand.java]: Why do you use `CacheCommandList`? Diagnostics doen't relates with `--cache` commands

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 15]
```diff
@@ -106,63 +84,79 @@
...
+                    nodeIds.addAll(argIter.nextStringSet(""));
+
+                    break;
+                case PATH:
+                    path = argIter.nextArg("");
+
+                    break;
+                default:
+                    throw new IllegalArgumentException(
+                        "Unexpected argumetn:" + arg + ", supported:" + Arrays.toString(PageLocksCommandArg.values()));
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: Please move closing parenthesis to the new line

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 16]
```diff
@@ -174,43 +168,67 @@ private void printResult(Map<ClusterNode, VisorPageLocksResult> res) {
...
+        ) {
             this.op = op;
-            this.type = type;
             this.filePath = filePath;
             this.allNodes = allNodes;
             this.nodeIds = nodeIds;
         }
     }
+
+    private enum PageLocksCommandArg implements CommandArg{
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: missing space before `{`

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 17]
```diff
@@ -106,63 +84,79 @@
...
     @Override public void printUsage() {
-        logger.info("View pages locks state information on the node or nodes.");
-        logger.info(CommandLogger.join(" ",
-            UTILITY_NAME, DIAGNOSTIC, PAGE_LOCKS, DUMP,
-            "[--path path_to_directory] [--all|--nodes nodeId1,nodeId2,..|--nodes consistentId1,consistentId2,..]",
-            "// Save page locks dump to file generated in IGNITE_HOME/work directory."));
-        logger.info(CommandLogger.join(" ",
-            UTILITY_NAME, DIAGNOSTIC, PAGE_LOCKS, DUMP_LOG,
-            "[--all|--nodes nodeId1,nodeId2,..|--nodes consistentId1,consistentId2,..]",
+        CommandLogger.log("View pages locks state information on the node or nodes.");
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: Please use static imports here

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java [hunk 18]
```diff
@@ -174,43 +168,67 @@ private void printResult(Map<ClusterNode, VisorPageLocksResult> res) {
...
-        /**
-         *
-         */
+        /** */
         private final Set<String> nodeIds;
 
         /**
          * @param op Operation.
-         * @param type Type.
          * @param filePath File path.
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/diagnostic/PageLocksCommand.java]: missing javadoc for `allNodes`

File: modules/core/src/main/java/org/apache/ignite/internal/visor/diagnostic/Operation.java [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.visor.diagnostic;
+
+/** */
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/diagnostic/Operation.java]: Please add class description

File: modules/core/src/main/java/org/apache/ignite/internal/visor/diagnostic/VisorPageLocksResult.java [hunk 20]
```diff
@@ -47,30 +48,13 @@ public VisorPageLocksResult(String payload) {
...
-        int length = in.readInt();
-
-        if (length != 0) {
-            byte[] bytes = new byte[length];
-
-            in.read(bytes);
-
-            payload = new String(bytes);
-        }
+        U.readString(in);
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/diagnostic/VisorPageLocksResult.java]: Missing assignment to `payload`?

File: modules/core/src/main/java/org/apache/ignite/internal/visor/diagnostic/VisorPageLocksTask.java [hunk 21]
```diff
@@ -110,25 +114,23 @@ private VisorPageLocksTrackerJob(VisorPageLocksTrackerArgs arg, boolean debug) {
         @Override protected VisorPageLocksResult run(VisorPageLocksTrackerArgs arg) {
             PageLockTrackerManager lockTrackerMgr = ignite.context().cache().context().diagnostic().pageLockTracker();
 
-            String op = arg.type();
-
             String result;
 
-            if ("dump".equals(op)) {
+            if (DUMP_FILE == arg.operation()) {
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/diagnostic/VisorPageLocksTask.java]: I think switch/case looks better here.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/diagnostic/VisorPageLocksTask.java [hunk 22]
```diff
@@ -138,4 +140,5 @@ else if ("dump_log".equals(op)) {
             return S.toString(VisorPageLocksTrackerJob.class, this);
         }
     }
+
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/diagnostic/VisorPageLocksTask.java]: Empty line mustn't be here

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 23]
```diff
@@ -129,6 +133,46 @@
...
+            + separatorChar + DEFAULT_TARGET_FOLDER + separatorChar);
+
+        customDiagnosticDir = new File(U.defaultWorkDirectory()
+            + separatorChar + "diagnostic_test_dir" + separatorChar);
+    }
+
+    /**
+     *
+     */
+    private void cleanDiagnosticDir(){
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: Missing space before `{` and comment must be single line

**Comment by dspavlov** [comment]
The ticket is resolved, closing PR.  
Feel free to reopen this PR if it is anyway actual.

**Pull request closed without merging**